File: frontend/src/app/services/local/models/cliente.ts
import { Endereco } from './endereco';
import { Conta } from './conta';

export type EstadoCliente = 'AGUARDANDO' | 'APROVADO' | 'RECUSADO';

export interface Cliente {
  nome: string;
  email: string;
  cpf: string;
  endereco: Endereco;
  telefone: string;
  salario: number;
  estado: EstadoCliente;
  dadosConta?: Conta;
  senha?: string; 
  gerenteCpf?: string;
  motivoRecusa?: string;
}

File: backend/cliente/src/main/java/dac/ufpr/cliente/entity/Cliente.java
package dac.ufpr.cliente.entity;

import dac.ufpr.cliente.enums.EnStatusCliente;
import dac.ufpr.cliente.enums.EnStatusIntegracao;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;


@Entity
@Table(name = "cliente", schema = "cliente")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Cliente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String nome;

    String cpf;

    String email;

    String telefone;

    BigDecimal salario;

    String endereco;

    String CEP;

    String cidade;

    String estado;

    @Enumerated(EnumType.STRING)
    EnStatusCliente status;

    String motivoRejeicao;

    LocalDateTime data_alteracao;

    long gerente_id;

}
